#VRML_SIM R2021b utf8
WorldInfo {
  coordinateSystem "NUE"
  contactProperties [
    ContactProperties {
      coulombFriction [
        1.5
      ]
    }
  ]
}
Viewpoint {
  orientation 0.0021609735533607345 -0.9546736783539247 -0.2976465992607489 3.175918145008285
  position 0.003658214063718024 2.5441136765758956 -3.3107746128542774
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 10 10
}
Robot {
  translation 0 0.14 0
  rotation 0 1 0 1.01503e-06
  children [
    InertialUnit {
      name "imu"
    }
    GPS {
    }
    DEF RF_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.2 0 0.2
      }
      device [
        RotationalMotor {
          name "rf_dir_motor"
        }
      ]
      endPoint Solid {
        translation 0.2 0 0.2
        children [
          DEF RF_RUN HingeJoint {
            jointParameters HingeJointParameters {
              axis 0 0 1
            }
            device [
              RotationalMotor {
                name "rf_run_motor"
                maxVelocity 100
              }
            ]
            endPoint Solid {
              rotation 1 0 3.9252239376397585e-17 1.5707999999999998
              children [
                DEF DRI Shape {
                  appearance PBRAppearance {
                    baseColor 1 0.411765 0.647059
                    metalness 0
                    emissiveColorMap ImageTexture {
                      url [
                        "C:/Users/zxy/Pictures/Default.jpg"
                      ]
                    }
                  }
                  geometry Cylinder {
                    height 0.03
                    radius 0.06
                  }
                }
              ]
              boundingObject USE DRI
              physics Physics {
              }
            }
          }
          DEF DIR Shape {
            appearance PBRAppearance {
              baseColor 1 0.976471 0.701961
              metalness 0
            }
            geometry Cylinder {
              height 0.03
              radius 0.05
            }
          }
        ]
        name "solid(1)"
        boundingObject USE DIR
        physics Physics {
        }
      }
    }
    DEF LF_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.2 0 -0.2
      }
      device [
        RotationalMotor {
          name "lf_dir_motor"
        }
      ]
      endPoint Solid {
        translation 0.2 0 -0.2
        children [
          DEF LF_RUN HingeJoint {
            jointParameters HingeJointParameters {
              axis 0 0 1
            }
            device [
              RotationalMotor {
                name "lf_run_motor"
                maxVelocity 100
              }
            ]
            endPoint Solid {
              rotation 1 7.850447875279512e-17 3.5327015438757803e-16 1.5708000000000013
              children [
                DEF DRI Shape {
                  appearance PBRAppearance {
                    baseColor 1 0.411765 0.647059
                    metalness 0
                    emissiveColorMap ImageTexture {
                      url [
                        "C:/Users/zxy/Pictures/Default.jpg"
                      ]
                    }
                  }
                  geometry Cylinder {
                    height 0.03
                    radius 0.06
                  }
                }
              ]
              boundingObject USE DRI
              physics Physics {
              }
            }
          }
          DEF DIR Shape {
            appearance PBRAppearance {
              baseColor 1 0.976471 0.701961
              metalness 0
            }
            geometry Cylinder {
              height 0.03
              radius 0.05
            }
          }
        ]
        name "solid(3)"
        boundingObject USE DIR
        physics Physics {
        }
      }
    }
    DEF LB_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.2 0 -0.2
      }
      device [
        RotationalMotor {
          name "lb_dir_motor"
        }
      ]
      endPoint Solid {
        translation -0.2 0 -0.2
        children [
          DEF LB_RUN HingeJoint {
            jointParameters HingeJointParameters {
              axis 0 0 1
            }
            device [
              RotationalMotor {
                name "lb_run_motor"
                maxVelocity 100
              }
            ]
            endPoint Solid {
              rotation 1 -1.9626119688198792e-16 3.532701543875782e-16 1.5707999999999998
              children [
                DEF DRI Shape {
                  appearance PBRAppearance {
                    baseColor 1 0.411765 0.647059
                    metalness 0
                    emissiveColorMap ImageTexture {
                      url [
                        "C:/Users/zxy/Pictures/Default.jpg"
                      ]
                    }
                  }
                  geometry Cylinder {
                    height 0.03
                    radius 0.06
                  }
                }
              ]
              boundingObject USE DRI
              physics Physics {
              }
            }
          }
          DEF DIR Shape {
            appearance PBRAppearance {
              baseColor 1 0.976471 0.701961
              metalness 0
            }
            geometry Cylinder {
              height 0.03
              radius 0.05
            }
          }
        ]
        name "solid(4)"
        boundingObject USE DIR
        physics Physics {
        }
      }
    }
    DEF RB_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.2 0 0.2
      }
      device [
        RotationalMotor {
          name "rb_dir_motor"
        }
      ]
      endPoint Solid {
        translation -0.2 0 0.2
        children [
          DEF RB_RUN HingeJoint {
            jointParameters HingeJointParameters {
              axis 0 0 1
            }
            device [
              RotationalMotor {
                name "rb_run_motor"
                maxVelocity 100
              }
            ]
            endPoint Solid {
              rotation 1 -7.85044787527952e-17 7.85044787527952e-17 1.570799999999999
              children [
                DEF DRI Shape {
                  appearance PBRAppearance {
                    baseColor 1 0.411765 0.647059
                    metalness 0
                    emissiveColorMap ImageTexture {
                      url [
                        "C:/Users/zxy/Pictures/Default.jpg"
                      ]
                    }
                  }
                  geometry Cylinder {
                    height 0.03
                    radius 0.06
                  }
                }
              ]
              boundingObject USE DRI
              physics Physics {
              }
            }
          }
          DEF DIR Shape {
            appearance PBRAppearance {
              baseColor 1 0.976471 0.701961
              metalness 0
            }
            geometry Cylinder {
              height 0.03
              radius 0.05
            }
          }
        ]
        name "solid(2)"
        boundingObject USE DIR
        physics Physics {
        }
      }
    }
    DEF BODY Solid {
      children [
        DEF BODY Shape {
          appearance PBRAppearance {
            baseColor 0.454902 0.552941 1
            metalness 0
            emissiveColorMap ImageTexture {
              url [
                "D:/Webots_Project/Wb_Steering_Wheel/Arrow.png"
              ]
            }
          }
          geometry Box {
            size 0.4 0.04 0.4
          }
        }
      ]
      boundingObject USE BODY
      physics Physics {
      }
    }
  ]
  boundingObject USE BODY
  physics Physics {
  }
  controller "my_controller"
}
